### PR TITLE
Update CONTRIBUTORS.md and supporting information

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary
This brings a much needed update to our contributor documentation. We're seeing a lot of third party contributors to our repository. Many of these bug fixes are great, but some of them are doing things we want to discourage. This tries to outline what those things are, and gives examples of them.

It also creates a new simple recommendation for making a change that goes against the rules, by filing a Fish Improvement Proposal (FIP).

- Updates CONTRIBUTING.md
- Adds a new FIP proposal issue template
- Updates README.md to mention CONTRIBUTING.md

## Testing Plan
No tests needed

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -55,32 +29,27 @@ If you don't know what contribution you can work on, here are a few suggestions:
...
-
-## Style Guide
-
-Iron Fish uses `eslint` and `prettier` to maintain consistent formatting on the TypeScript codebase.
-For the Rust codebase, we are using `rustfmt`.
+If you are working on something in one of these categories, we will not accept your PR if you don't open a Fish Improvement Proposal (FIP).
 
-Please run it before submitting a change.
+ - Upgrading Package Versions
+   - This is a common attack vector by malicious agents. We lock down our packages for this very reason, and only allolw the core team to upgrade packages unlesse you submit a FIP first explaining why you want to upgrade the package.
```
Comment (deekerno) [inline CONTRIBUTING.md:37]: Nit: "allolw" -> "allow", "unlesse" -> "unless"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -55,32 +29,27 @@ If you don't know what contribution you can work on, here are a few suggestions:
...
+ - Upgrading Node Versions
+   - There is a large impact  in upgrading node versions. Iron Fish takes advantage of many experimental node features that at some are not fully fleshedout. We use workers, and native code boundries. Node is also known to introduce bugs in newer versions, even LTS. Because of this, there is production testing we do on our side when upgrading node to ensure it's compatible. Because of this, we don't allow usersc to upgrade our node versions unless a FIP is filed.
```
Comment (deekerno) [inline CONTRIBUTING.md:39]: Nit: Extra space after "large impact"; "fleshedout" -> "fleshed out", "boundries" -> "boundaries", "usersc" -> "users"

Also, consider "...experimental node features, some of which are not fully fleshed out."

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -55,32 +29,27 @@ If you don't know what contribution you can work on, here are a few suggestions:
...
+ - Refactor a core system
+   - Often engineers have visions for core systems, and may be in the process of changing them. FIP's help avoid overlap and to have your changes over-written. Some core systems include, MerkleTree, Blockchain, PeerNetwork, Verifier, Consensus.
```
Comment (deekerno) [inline CONTRIBUTING.md:41]: Consider "...avoid overlap and having your changes over-written".

Nit: "Some core systems include..." (remove first comma)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -55,32 +29,27 @@ If you don't know what contribution you can work on, here are a few suggestions:
...
+ - Making new product design decisions that has no precident
```
Comment (deekerno) [inline CONTRIBUTING.md:42]: Nit: "precident" -> "precedent"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -55,32 +29,27 @@ If you don't know what contribution you can work on, here are a few suggestions:
...
+   - This one is more complicated. If our products works one way, and you open a PR to change the core product principals to work in another way, then we are going to close your PR as it does not fit into our vision. Try to match existing precident as much as possible in your code.
```
Comment (deekerno) [inline CONTRIBUTING.md:43]: Nit: "principles" should be used; "precident" -> "precedent"

**Review by deekerno — CHANGES_REQUESTED** [review]
Just a few nits; exciting changes overall!

File: CONTRIBUTING.md (hunk 5 above)
Comment (NullSoldier) [reply CONTRIBUTING.md:43]: I think for these text changes it's easier if you just use the suggestion feature of github.

Comment (deekerno) [reply CONTRIBUTING.md:43]: Oh, I've actually never seen that button until now. I'll propose suggestions instead.

Comment (deekerno) [reply CONTRIBUTING.md:43]:
```suggestion
   - This one is more complicated. If our products works one way, and you open a PR to change the core product principles to work in another way, then we are going to close your PR as it does not fit into our vision. Try to match existing precedent as much as possible in your code.
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (deekerno) [reply CONTRIBUTING.md:42]:
```suggestion
 - Making new product design decisions that has no precedent
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (deekerno) [reply CONTRIBUTING.md:41]:
```suggestion
   - Often engineers have visions for core systems, and may be in the process of changing them. FIP's help avoid overlap and having your changes over-written. Some core systems include MerkleTree, Blockchain, PeerNetwork, Verifier, Consensus.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (deekerno) [reply CONTRIBUTING.md:37]:
```suggestion
   - This is a common attack vector by malicious agents. We lock down our packages for this very reason, and only allow the core team to upgrade packages unless you submit a FIP first explaining why you want to upgrade the package.
```

File: CONTRIBUTING.md (hunk 5 above)
Comment (NullSoldier) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (NullSoldier) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 1 above)
Comment (NullSoldier) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (NullSoldier) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (NullSoldier) [reply CONTRIBUTING.md]: Fixed

**Review by deekerno — APPROVED** [review]

**Pull request merged**